### DISCUSSION TITLE
What to put in version 3?

### DISCUSSION THREAD

**Discussion by marocchino** [ffaker/ffaker#486] [General]
### To do

Please let me know if there is anything more to add or subtract from here:
- rbi or similar type support: I would like autocompletion and type inference to be possible at least in the largest lsp.
- Documentation with yard or rbdocs: this one came from README
- Major changes affecting usability
- Config strategy
- Tear down utils: Now that the module is being imported, I feel that the module's method set is more than necessary. I want to end with a reference to the part with little dependence. 
- Clean up code now marked as deprecated

### Schedule

If there are no other issues, we will start working by splitting branches right after the config is complete. 
I'd like to include the following and I'm thinking of a May-June release. Even in the worst case, I want to release it later this year. 

### What to discuss

- Distribution of work
- Materialization of the list of major changes
- Add or change to-dos 

So, I would appreciate your opinion and Emoji's reactions.

**Comment by Volosh1n** [comment]
Use safe emails (those ones, which ends with '@example.com') as `FFaker::Internet.safe_email`, while `FFaker::Internet.email` should just be alias to safe_email. Also, unsafe_email should generate emails with real providers, as gmail.com, yahoo.com, etc. This feature will close #483. Also, we can provide 2.x.x version with warning about such change. Just a suggestion, maybe such things could be configured in future config strategy

**Comment by Volosh1n** [comment]
#453 could be fixed too
